content/docs/latest/setup/debug/install-debugging-tools.md

```diff
@@ -125,3 +125,40 @@ Press ^] three times within 1s to kill container.
 [root@srv-3qy0p ~]# dnf -y install emacs-nox
 [root@srv-3qy0p ~]# emacs /media/root/etc/systemd/system/newapp.service
 ```
+
+## Early boot debugging over console
+
+When you need an interactive shell inside the initramfs before the system boots (for example to inspect units or mounts before systemd starts userspace), you can drop into rd.break and direct all I/O to the serial console only.
+
+This kernel command line disables VGA output and uses the serial port as the sole interactive console:
+
+```
+rd.break rd.shell rd.debug loglevel=7 console=ttyS0,115200n8 earlycon=ttyS0,115200 systemd.show_status=yes systemd.log_level=debug systemd.log_target=console
+```
+
+Notes:
+- Only specify a single console= to ensure the serial port is the exclusive input console. If your platform injects console=tty0 by default, remove it so the display is not used.
+- Replace ttyS0,115200n8 with the device/speed appropriate for your platform:
+  - PC/QEMU (16550A): ttyS0,115200n8 (earlycon=ttyS0,115200)
+  - Virtio console: hvc0 (earlycon=hvc0; speed not required)
+  - Many ARM platforms: ttyAMA0,115200 (earlycon=ttyAMA0,115200)
+- Some cloud providers (e.g., Scaleway) require that you explicitly configure the instance to expose the serial console and that you pass the correct console= parameter for their environment. Consult the provider’s documentation for the exact device name.
+- rd.break opens a shell in the initramfs; rd.shell also drops you to a shell if the initramfs hits an error. rd.debug and loglevel=7 increase verbosity. systemd.show_status=yes and systemd.log_target=console print systemd status and logs to the console early, which is particularly useful when services fail to start without visible errors on the default console.
+- Security: these options yield an unauthenticated root shell in early boot. Remove them once you have finished debugging.
+
```

Comment:
For completeness: The new Alpha has a minimal first stage initrd that has its own debug arguments `rd.earlytrace` and `rd.earlyshell` which do something different. Since the minimal initrd is just about loading `/usr` I think this is only needed for debugging when drivers are missing.